---
title: "BIOMI6300_Project1:OMV_Analysis"
author: "Jiucheng Ding"
date: "`r Sys.Date()`"
output: html_document
---

#File setup:
```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE)
```


#Adding Introduction:
  Outer membrane vesicles (OMVs) are derived from Gram-negative bacteria, are formed through a budding and release mechanism on the outer membrane.  The spheres are ranged between 20-200 nm in diameter, encapsulated mostly by lipids.They contain bioactive proteins and other components and thus may play biological roles to the neighboring microbiome community and to the host. 
  Although the production of OMVs have been well-reported and documented, the mechanism behind is still not quite clear; investigation on their formation mechanisms and bioactive roles is still required. This study tracked the growth and vesiculation values of the whole genome knockout library of Escherichia coli mutant strains through a high-throughput experimental method and systems-scale analysis. The mutations were reported to truncate the structure of the OMVs and led to hypervesiculation, and mutants in oxidative stress perform lowered level of vesiculation in E.coli.
  
  
  
#Install tidyverse without installation message:
```{r packages}
library(tidyverse)
library(dp)
```

#Loading data of OMV production values:
```{r data}
vesiculation<- read_csv("../Data/vesiculation.csv")
```
#Perfrom some analysis with this OMV production data:
#The data file shows the difference in OMV production of the gene-knockout library comparing to their normal state.
```{r data}
summarize(vesiculation, 
          averageLogOMV=mean(average.log.OMV), 
          averageAntilogOMV=mean(average.antilog.omv),
          averageSTDEVAntilog=mean(stdev.antilog))
```

#Then make scatterplot to show the distribution of OMV production change (linear) for all samples, with a few outliers removed.
```{r data}
ggplot(data=vesiculation) +
  aes(x=bID) + 
  labs(x= "E.coli K12 unique identifier") +
  aes(y=average.antilog.omv) +
  ylim(-1,8)+
  labs(y="Average OMV Production Change (Folds)")+
  geom_point(size=0.20,alpha=0.6)+
  geom_hline(yintercept=1, color="blue", linetype="solid", size=1)+
  geom_hline(yintercept=1.091811, color="orange",linetype="dashed", size=0.7)
  labs(title = "Average impact on OMV production (Antilogged, Linear)")
```
Figure1: This scatterplot shows the OMV production change in antilog, linear. Each dot represents an individual sample.A few outliers above 8 folds are ignored. The blue solid has a y-intercept of 1, indicating that the values above this line have an increased OMV production; those below represent those with lowered OMV production.


#Let's make another figure with same data, but in log10 scale.
```{r data}
ggplot(data=vesiculation) +
  aes(x=bID) + 
  labs(x= "E.coli K12 unique identifier") +
  aes(y=average.log.OMV) +
  labs(y="Average OMV Production Change (log10 of Folds)")+
  geom_point(size=0.25, alpha=0.6)+
  geom_hline(yintercept=0, color="blue", linetype="solid", size=1)+
  geom_hline(yintercept=-0.02552768, color="orange",linetype="dashed", size=0.7)
  labs(title = "Average impact on OMV production (Log10)")
```
Figure2:This scatterplot shows the OMV production change in log10. Each dot represents an individual sample.The blue solid has a y-intercept of 0, indicating that the values above this line have an increased OMV production; those below represent those with lowered OMV production.







#To determine the session info:
install.packages("devtools")
library(devtools)
session_info()

